kolibri/core/discovery/utils/network/search.py

```diff
@@ -130,31 +144,69 @@ def remove_service(self, zeroconf, type, name):
             del self.instances[id]
 
 
-def get_available_instances(timeout=2, include_local=True):
+cache = Cache(os.path.join(KOLIBRI_HOME, "zeroconf_cache"))
+
+
+def get_peer_instances():
     """Retrieve a list of dicts with information about the discovered Kolibri instances on the local network,
     filtering out those that can't be accessed at the specified port (via attempting to open a socket)."""
     if not ZEROCONF_STATE["listener"]:
         initialize_zeroconf_listener()
         time.sleep(3)
-    instances = []
-    for instance in ZEROCONF_STATE["listener"].instances.values():
-        if instance["local"] and not include_local:
-            continue
-        if not _is_port_open(instance["ip"], instance["port"], timeout=timeout):
-            continue
-        instance["self"] = (
-            ZEROCONF_STATE["service"] and ZEROCONF_STATE["service"].id == instance["id"]
-        )
-        instances.append(instance)
-    return instances
+
+    return list(ZEROCONF_STATE["listener"].instances.values())
+
+
+def run_peer_discovery(timeout=2, include_local=True):
```

Comment:
This is the high level interface to peer discovery.  It handles caching and is debounced in order to prevent spamming the network.

The debounce time is currently set to 5 seconds (in `settings.py` via `ZEROCONF_MIN_ALLOWED_REFRESH`)

Reply:
As a rule we have tended to avoid adding non-Django settings to the settings files, and left things that need to be configured to things specified in `options.py`.

It seems like the main motivation here is to have differential behaviour between prod and develop - but I think it might be easier to just have that modified during dev time when someone is working on it.

I'd suggest just setting it within this file as `5` for now. If we want to make it configurable we can add it to `options.py`.

Reply:
(calling from multiple processes, on the other hand, would initialize a fresh listener for each, which adds a bit of network traffic as it pings to ask who's out there when it first starts up -- my top-level comment will make a suggestion for avoiding this completely)

Reply:
This is all taken care of now!  It works great to just let zeroconf handle all that.